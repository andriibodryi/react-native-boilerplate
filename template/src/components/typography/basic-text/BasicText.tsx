import { FC } from 'react';
import { Text } from 'react-native';

import { useBasicTextStyles } from './basic-text.presets';
import { BasicTextTypes } from './basic-text.types';

const BasicText: FC<BasicTextTypes> = ({ children, style, ...rest }) => {
  const styles = useBasicTextStyles();
  return (
    <Text style={[styles.root, style]} {...rest}>
      {children}
    </Text>
  );
};

export default BasicText;
