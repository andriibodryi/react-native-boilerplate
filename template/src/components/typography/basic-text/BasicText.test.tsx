import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import BasicText from './BasicText';

describe('BasicText', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<BasicText>basic text</BasicText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
