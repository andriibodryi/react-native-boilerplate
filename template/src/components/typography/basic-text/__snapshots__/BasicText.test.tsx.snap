// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicText should render correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4D4D4D",
        "fontFamily": "Aspira-Regular",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  basic text
</Text>
`;
